core/pkg/kubecost/window.go

```diff
@@ -28,8 +28,60 @@ var (
 	rfc3339             = `\d\d\d\d-\d\d-\d\dT\d\d:\d\d:\d\dZ`
 	rfcRegex            = regexp.MustCompile(fmt.Sprintf(`(%s),(%s)`, rfc3339, rfc3339))
 	timestampPairRegex  = regexp.MustCompile(`^(\d+)[,|-](\d+)$`)
+
+	tOffsetLock sync.Mutex
+	tOffset     *time.Duration
+
+	utcOffsetLock sync.Mutex
+	utcOffsetDur  *time.Duration
 )
 
+// get and cache the thanos offset duration.
+// TODO: Due to dependencies here, we have to drag a non-core config option into
+// TOOD: core scope. Any solution here would be a one-off until we can generalize
+// TODO: global configuration options.
+func thanosOffset() time.Duration {
```

Comment:
As mentioned above, Thanos has some leaked implementation into the core module. To avoid expanding the scope of this refactor, I'm punting due to the impact being low. We'll address these infrequent cases next pass.